### PR TITLE
HPCC-31872 Allow alternate certificate domains 

### PR DESCRIPTION
**Pull request by asselitx**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by asselitx** [comment]
@afishbeck Could you please review? One question is if the alternate domains also need to be applied to the spiffe URIs? I didn't see any other cases where the current domain is used to generate `dnsNames` entries.

File: helm/hpcc/templates/_helpers.tpl [hunk 1]
```diff
@@ -1929,7 +1930,9 @@ spec:
      {{- end }}
   dnsNames:
      {{- range $dnsName := $local.dnsNames }}
-  - {{ (printf "%s.%s" $dnsName $domain) | quote }}
+     {{- range $altDomain := $local.allDomains }}
```
Comment (afishbeckln) [inline helm/hpcc/templates/_helpers.tpl:1933]: If I were being super picky I would say the variable name "altDomain" is not right.  $local.allDomains contains the primary domain and the alts.  But hardly matters.

**Review by afishbeckln — APPROVED** [review]
@asselitx looks good.

**Comment by asselitx** [comment]
@ghalliday squashed and renamed PR to follow convention. The failing 'macos-14' check is happening across most PRs currently.

**Review by ghalliday — APPROVED** [review]

File: helm/hpcc/templates/_helpers.tpl [hunk 2]
```diff
@@ -1894,6 +1894,7 @@ args:
       {{- end -}}
      {{- end -}}
      {{- $_ := set $local "dnsNames" (uniq $local.dnsNames ) -}}
+     {{- $_ := set $local "allDomains" (prepend (default list $issuer.alternativeDomains) $domain ) -}}
```
Comment (ghalliday) [inline helm/hpcc/templates/_helpers.tpl:1897]: Minor: This the default operator is more normally written as

($issuer.alternativeDomains | default list)

(I had to read some documentation to check the semantics...)

**Pull request merged**